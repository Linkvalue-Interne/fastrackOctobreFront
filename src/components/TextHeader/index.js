import React from 'react';
import TextHeader from '../commons/TextHeader';

const Title = (props) => (
  <TextHeader title={props.title} subtitle={props.subtitle} />
);

export default Title;
