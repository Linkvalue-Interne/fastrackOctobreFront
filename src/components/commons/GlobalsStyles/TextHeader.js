import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const Header = styled.div`
  width: calc(100% - 80px);
  margin-left: 80px;
  display: flex;
  height: 5vh;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding-top: 2.5rem;
  padding-bottom: 1.5rem;
`;

const Title = styled.h2`
  color: ${(props) => props.theme.colors.white};
  font-size: 2rem;
  font-weight: bold;
`;

const Subtitle = styled.h3`
  color: ${(props) => props.theme.colors.white};
  font-size: 1rem;
  font-weight: 300;
`;

const TextHeader = ({ title, subtitle }) => (
  <Header>
    <Title>{title}</Title>
    <Subtitle>{subtitle}</Subtitle>
  </Header>
);

TextHeader.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default TextHeader;
