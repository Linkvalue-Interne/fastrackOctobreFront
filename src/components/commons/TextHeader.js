import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const Header = styled.div`
  width: calc(100% - 80px);
  margin-left: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 2rem 0;
`;

const Title = styled.h2`
  color: #FFF;
  font-size: 2rem;
  font-weight: bold;
`;

const Subtitle = styled.h3`
  color: #FFF;
  font-size: 1rem;
  font-weight: 300;
`;

const TextHeader = (props) => (
  <Header>
    <Title>{props.title}</Title>
    <Subtitle>{props.subtitle}</Subtitle>
  </Header>
);

TextHeader.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

export default TextHeader;
