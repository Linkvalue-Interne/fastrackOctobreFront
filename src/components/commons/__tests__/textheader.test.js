import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import TextHeader from '../TextHeader';

configure({ adapter: new Adapter() });

describe('test', () => {
  it('render without crashing', () => {
    const test = shallow(<TextHeader title="test" subtitle="test"/>);
    expect(test).toMatchSnapshot();
  });
});
