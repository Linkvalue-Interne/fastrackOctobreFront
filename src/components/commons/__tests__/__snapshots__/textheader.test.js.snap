// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render without crashing 1`] = `
<Styled(div)>
  <Styled(h2) />
  <Styled(h3) />
</Styled(div)>
`;
